### DISCUSSION TITLE
ValueError: too many values to unpack (expected 5)

### DISCUSSION THREAD

**Discussion by jdiazram** [JavierBZ/Tarea3-IIC3697#3] [Q&A] [answered]
Hola Javier
Cuando corro el programa, me aparece este error _ValueError: too many values to unpack (expected 5)_, cuando la variable _t_ llega a 50000 en la linea de _obs_t, act_t, next_obs, rew_t, done_mask = replay_memory.sample(batch_size)_ . Alguna idea o ayuda de que puede estar pasando? revisé el resultado de replay_memory.sample y me retorna 32 valores del tipo:

[Transition(
state=<gym.wrappers.frame_stack.LazyFrames object at 0x7fe299c29dd0>, action=5, 
next_state=<gym.wrappers.frame_stack.LazyFrames object at 0x7fe299c29f50>, 
reward=0.0, 
done=False),....

Muchas gracias de antemano.

Saludos

**Comment by ptflores1** [comment, marked as answer]
Si estas hacieno que sample retorne una lista de 32 elementos entonces el error se produce porque intentas asignar los 32 elementos a las 5 variables `obs_t, act_t, next_obs, rew_t, done_mask`. Como está pensado el método sample, es que retorne 5 cosas, cada una con la cantidad de elementos segun el `batch_size`, por ej `act_t `puede ser una lista con 32 valores de acciones.

**Reply by JavierBZ** [reply]
Tiene razón tu compañero, lo que debe retornar la función sample son `obs_t, act_t, next_obs, rew_t, done_mask`, pero en formato batch size, es decir, arreglos cuya primera dimensión es la de batch.
Espero te sirva

**Reply by jdiazram** [reply]
Gracias...lo revisaré
